src/virt.c

```diff
@@ -2120,6 +2136,12 @@ static int add_interface_device(struct lv_read_state *state, virDomainPtr dom,
     return -1;
   }
 
+  type_copy = strdup(type);
+  if (!type_copy) {
+    sfree(type_copy);
```

Comment:
Should this be freeing type not type_copy?

Reply:
Should be freeing address_copy and path_copy if they exist.  Nice find.